src/mixins/name-request-mixin.ts

```diff
@@ -1,18 +1,59 @@
 // Libraries
 import axios from '@/utils/axios-auth'
 import { Component, Vue } from 'vue-property-decorator'
-import { NameRequestStates } from '@/enums/nameRequestStates'
+import { NameRequestStates, EntityTypes } from '@/enums'
+import { NameRequestIF } from '@/interfaces'
 
 /**
  * Name request mixin for processing NR responses
  */
 @Component
 export default class NameRequestMixin extends Vue {
+  /**
+   * Generate name request state for the store
+   * @param nr the name request response payload
+   * @param filingId the filing id
+   */
+  generateNameRequestState (nr: any, filingId: number): NameRequestIF {
+    const approvedName = nr.names.filter(name => name.state === NameRequestStates.APPROVED)[0].name
```

Comment:
You could also use `find`, which returns a single entry (or undefined). Theoretically it's faster because it will return when it finds the first entry, instead of filtering the entire array.

Reply:
Updated! Thanks!